Library/Homebrew/utils.rb

```diff
@@ -770,3 +770,24 @@ def number_readable(number)
   (numstr.size - 3).step(1, -3) { |i| numstr.insert(i, ",") }
   numstr
 end
+
+def truncate_text_to_approximate_size(s, max_bytes, options = {})
+  front_weight = options[:front_weight] || 0.5
+  glue = options[:glue] || "\n[...snip...]\n"
+  return s if s.bytesize <= max_bytes
+
+  if front_weight < 0 || front_weight > 1
+    raise "opts[:front_weight] must be between 0.0 and 1.0"
+  end
+  scale = 0.99 # just a tad small to allow for the glue
+  while true
```

Comment:
> Another option is reencoding this in a fixed-length encoding.

Aw, crap, this doesn't actually work: the worst-case scenario for variable-length encodings may be longer than fixed-length encodings. That's part of the fixed/variable trade-off. Concrete example: UTF-32 is 4 bytes per character, but UTF-8 may be up to 8 bytes per character. Fundamentally, truncating in one encoding can't guarantee you a max byte size in another encoding unless both encodings are fixed-length.